hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java

```diff
@@ -106,13 +129,95 @@ public void doGenesisSetup(@NonNull final Dispatch dispatch) {
+
+        // And then we update the system files for fees schedules, throttles, override properties, and override
+        // permissions from any upgrade files that are present in the configured directory
+        final var config = dispatch.config();
+        final var filesConfig = config.getConfigData(FilesConfig.class);
+        final var adminConfig = config.getConfigData(NetworkAdminConfig.class);
+        final List<AutoSysFileUpdate> autoUpdates = List.of(
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.feeSchedules(), config),
+                        adminConfig.upgradeFeeSchedulesFile(),
+                        SystemSetup::parseFeeSchedules),
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.throttleDefinitions(), config),
+                        adminConfig.upgradeThrottlesFile(),
+                        SystemSetup::parseThrottles),
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.networkProperties(), config),
+                        adminConfig.upgradePropertyOverridesFile(),
+                        in -> parseConfig("override network properties", in)),
+                new AutoSysFileUpdate(
+                        createFileID(filesConfig.hapiPermissions(), config),
+                        adminConfig.upgradePermissionOverridesFile(),
+                        in -> parseConfig("override HAPI permissions", in)));
+        autoUpdates.forEach(update -> {
+            if (update.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
+                dispatch.stack().commitFullStack();
+            }
+        });
+    }
+
+    /**
+     * Encapsulates logic to attempt the automatic update of a system file by parsing the new contents of the
+     * corresponding file from the post-upgrade location.
+     *
+     * @param fileId the file ID of the system file
+     * @param upgradeFileName the name of the upgrade file
+     * @param upgradeFileParser the function to parse the upgrade file
+     */
+    private record AutoSysFileUpdate(
+            @NonNull FileID fileId,
+            @NonNull String upgradeFileName,
+            @NonNull Function<InputStream, Bytes> upgradeFileParser) {
+        /**
+         * Attempts to update the system file using the given system context if the corresponding upgrade file is
+         * present at the given location and can be parsed with this update's parser.
+         *
+         * @param postUpgradeLoc the location of the post-upgrade files
+         * @param systemContext the system context
+         * @return {@code true} if a synthetic update was dispatched, {@code false} otherwise
+         */
+        boolean tryIfPresent(@NonNull final String postUpgradeLoc, @NonNull final SystemContext systemContext) {
+            final var path = Paths.get(postUpgradeLoc, upgradeFileName);
+            if (!Files.exists(path)) {
+                log.info(
```

Comment:
Isn't this a warn or error ?

Reply:
Used `INFO` because there's no real need to replace these files at most upgrade boundaries.